java-generator/core/src/main/java/io/fabric8/java/generator/nodes/AbstractJSONSchema2Pojo.java

```diff
@@ -69,11 +81,17 @@ protected String getClassType() {
     return getType();
   }
 
-  protected AbstractJSONSchema2Pojo(Config config, String description, final boolean isNullable, JsonNode defaultValue) {
+  protected AbstractJSONSchema2Pojo(Config config, String description, final boolean isNullable, JsonNode defaultValue,
+      final Optional<ValidationProperties> validationProperties) {
```

Comment:
Please, never use Optionals as method parameters, use them only as return types.

The reasoning is very simple, consider I invoke the constructor with `super(theConfig, theDescription, theIsNullable, theDefaultalue, null)`, the NPE problem would still be there.

If you're willing to do this with an Optional, then the suggestion would be to keep the parameter as `ValidationProperties validationProperties`,
and then replace your constructor body with:
```java
Optional.ofNullable(values).ifPresent(values -> {
//...
```
Or better and simpler just use an if not null clause.

Reply:
+1, thanks @manusa - fixing.